### PR TITLE
add email formatter

### PR DESCRIPTION
**Pull request by schiessle**
add email formatter to create mailto-links for email addresses, used here: https://github.com/nextcloud/server/pull/1998

cc @nickvergessen

### CODE REVIEW

**Comment by schiessle** [comment]
@nickvergessen any ideas regarding the failing test? Can't reproduce this locally.

**Comment by schiessle** [comment]
ok, it seems like it failed because of the "sharedialogmailview.js" we forget to remove. I can reproduce it on master and on https://github.com/nextcloud/server/pull/1998 the tests passed because of https://github.com/nextcloud/server/pull/1998/commits/be862ee2ed7040704adfe69b67b382c29dca0d83

So we can ignore the failing tests here...

On the

File: lib/Formatter/EmailFormatter.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	/**
+	 * create a mailto link
+	 *
+	 * @param IEvent $event
+	 * @param string $parameter The parameter to be formatted
+	 * @return string The formatted parameter
+	 */
+	public function format(IEvent $event, $parameter) {
+		$displayName = $this->getDisplayName($parameter);
+		return '<a href="mailto:' . $parameter . '">' . Util::sanitizeHTML($displayName) . '</a>';
```
Comment (LukasReschke) [inline lib/Formatter/EmailFormatter.php]: This introduces a XSS vector. Make sure to escape $parameter as well or this leads to a reflected XSS 🙈

Comment (LukasReschke) [inline lib/Formatter/EmailFormatter.php]: screen shot 2016-11-03 at 22 02 31

Comment (LukasReschke) [inline lib/Formatter/EmailFormatter.php]: On another remark: `unsafe-eval` has to die 🔫 
(https://github.com/nextcloud/server/issues/1185)

Comment (schiessle) [inline lib/Formatter/EmailFormatter.php]: fixed

Comment (schiessle) [inline lib/Formatter/EmailFormatter.php]: @nickvergessen I don't see a problem with hard-coded HTML. We just need to probably escape the parameters which I have done with my last changes

**Comment by MorrisJobke** [comment]
@schiessle This doesn't generate the email links for me :8

**Comment by nickvergessen** [comment]
Yeah code changed completly, will make a new PR

**Pull request closed without merging**